SCAutolib/models/gui.py

```diff
@@ -0,0 +1,304 @@
+from SCAutolib import run, logger
+
+import os
+from time import sleep, time
+import cv2
+import numpy as np
+import pandas as pd
+from io import StringIO
+import pytesseract
+import keyboard
+import uinput
+import functools
+
+
+class Screen():
+    """Captures the screenshots."""
+
+    def __init__(self, directory: str):
+        self.directory = directory
+        self.screenshot_num = 1
+
+    def screenshot(self, timeout: float = 30):
+        """Runs ffmpeg to take a screenshot.
+
+        :return: filename of the screenshot
+        :rtype: str
+        """
+
+        logger.debug(f"Taking screenshot number {self.screenshot_num}")
+
+        filename = f'{self.directory}/{self.screenshot_num}.png'
+        t_end = time() + timeout
+        captured = False
+
+        while time() < t_end and not captured:
```

Comment:
What is reasoning behind this while? Why do we need to repeat this action? It deserves at least a comment to explain.